pkg/controller/deployment/deployment_controller.go

```diff
@@ -410,6 +411,13 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	if d.Spec.Rollback != nil {
+		revision := d.Spec.Rollback.Revision
+		if err := dc.rollback(d, revision); err != nil {
```

Comment:
This updates the deployment (by removing the rollbackConfig) and later while syncing you try to update (note the deployment will have the same resource version as prior to the rollback update) again by adding the revision annotation which most probably will result in a conflict. I think you should return here.